client/src/types/settings.ts

```diff
@@ -11,6 +11,15 @@ export enum KeyBinding {
 
 export type LineNumber = 'on' | 'off' | 'relative' | 'interval'
 export type CursorBlinking = 'blink' | 'smooth' | 'phase' | 'expand' | 'solid'
+export type CursorStyle =
```

Comment:
We should  use `enum` whenever we are creating variants. For example:

```
enum Gender {
 MALE = 'male',
 FEMALE = 'female',
 OTHER = 'other',
}
```
instead of 
```
type Gender = 'male' | 'female' | 'other'
```